lib/pipeline/assets/promise.yaml

```diff
@@ -14,13 +14,13 @@ spec:
     apiVersion: apiextensions.k8s.io/v1
     kind: CustomResourceDefinition
     metadata:
-      name: redis.marketplace.kratix.io
+      name: custom-namespaces.marketplace.kratix.io
     spec:
       group: marketplace.kratix.io
       names:
-        kind: redis
-        plural: redis
-        singular: redis
+        kind: custom-namespace
+        plural: custom-namespaces
+        singular: custom-namespace
```

Comment:
Am I correct that this will mean that we have both (this and bash/bashes) our test data now having different singular and plurals?

I would prefer to see us cover both scenarios. That being said, it would be pretty cryptic that we were doing that, so it may require some decent descriptions in either the variable names or comments!

Reply:
These assets are only used in the pipeline unit tests, so I think just having the plural here is fine (since that's the field we want to have passed through) - we should definitely cover both scenarios in the system tests though!

In the system tests, we have some coverage of both:
- Different plural with `bash`/`bashes`
- Identical plural with `redis`/`redis` in the promise-with-requirement tests ([here](https://github.com/syntasso/kratix/blob/main/test/system/system_test.go#L145))

We could probably still make the system test coverage better (i.e. I'm not sure we cover all of the same assertions in each scenario) but that might be one for a separate PR!